pkg/lightning/restore/meta_manager.go

```diff
@@ -525,9 +543,34 @@ func (m *dbTaskMetaMgr) InitTask(ctx context.Context) error {
 		DB:     m.session,
 		Logger: log.L(),
 	}
-	// avoid override existing metadata if the meta is already inserted.
-	stmt := fmt.Sprintf(`INSERT IGNORE INTO %s (task_id, status) values (?, ?)`, m.tableName)
-	err := exec.Exec(ctx, "init task meta", stmt, m.taskID, taskMetaStatusInitial.String())
+
+	err := exec.Transact(ctx, "check and init task status", func(ctx context.Context, tx *sql.Tx) error {
+		// avoid override existing metadata if the meta is already inserted.
+		stmt := fmt.Sprintf(`INSERT IGNORE INTO %s (task_id, status) values (?, ?)`, m.tableName)
```

Comment:
Why insert be fore select......

Reply:
Ensure there is exact one row represent current lightning. Need the reset the task status if current task's status is `exit_unfinished`